model/lifecycle.go

```diff
@@ -54,6 +54,50 @@ type VersionToRestart struct {
 	TaskIds   []string `json:"task_ids"`
 }
 
+func UpdateVersionandPatchStatusForBuild(buildIds []string) error {
+	builds, err := build.Find(build.ByIds(buildIds))
+	if err != nil {
+		return errors.Wrapf(err, "fetching builds")
+	}
+	for _, build := range builds {
+		buildStatusChanged, err := UpdateBuildStatus(&build)
+		if err != nil {
+			return errors.Wrapf(err, "updating build '%s' status", build.Id)
+		}
+		// If no build has changed status, then we can assume the version and patch statuses have also stayed the same.
+		if !buildStatusChanged {
+			continue
+		}
+
+		buildVersion, err := VersionFindOneId(build.Version)
```

Comment:
This may involve us updating version status n number of times if every build is for the same version. Instead, can we keep a map of versions to update outside of the loop, and then loop through those after we finish updating builds? (we can also update patches in that second loop)

Reply:
done